src/manager.cpp

```diff
@@ -199,6 +200,26 @@ static SubprocessHandle exec_cpp_module(const ModuleStartInfo& module_info, std:
     return handle;
 }
 
+static SubprocessHandle exec_rust_module(const ModuleStartInfo& module_info, std::shared_ptr<RuntimeSettings> rs) {
```

Comment:
There is really no point here in passing the `RuntimeSettings` as shared pointer, just pass it as a (ideally const) reference.

Reply:
At the moment this is just to keep it in line with the other exec_* functions, I would suggest we refactor this in a new PR

Reply:
Add a TODO then for now so we do not forget? Reference this MR for posterity.